plugins/catalog-backend-module-bitbucket-server/src/lib/BitbucketServerClient.ts

```diff
@@ -81,7 +82,13 @@ export class BitbucketServerClient {
       request,
       getBitbucketServerRequestOptions(this.config),
     );
-    return response.json();
+    try {
+      return await response.json();
+    } catch {
+      throw new NotFoundError(
+        `Malformed json response when attempting to get repository '${options.repo}' in project '${options.projectKey}' from Bitbucket Server. Repository might not exist.`,
```

Comment:
also the error looks very specific, meaning that it will always return 404 even when the error is not supposed to be 404, with `Malformed json response` as a message, even when the json response is not malformed. Is this really wanted? 🤔

Reply:
@vinzscam I was suggested previously to make it a "NotFoundError" since I was getting an error at that point when the repository did not exist. I can see how maybe there could be a permissions issue, but I would assume the BitbucketServerProvider would catch that much earlier. Therefore the only error I can foresee happening at this point is the repository not being found and therefore the `response.json()` causing an error. Do you have any suggestions for a different type of error to use?

Reply:
I think a permissions issue is totally possible here as well. The events entry point to backstage is unprotected, so in theory anybody can send whatever garbage into it if they are within the network. As long as enough details are logged here about what type of response from bitbucket API is being received, it should be good.

The types of errors I can imagine:
- Fake payloads that have nonsense data in it. The logs should say that the repo and what was being searched for so that the Backstage admin can figure whats going on
- Payloads for repos that the backstage bitbucket API key doesn't have access to. Perhaps a repo has the events hooked up to send to backstage but backstage bitbucket API key doesn't have read access. So the BB provider would not have created a location yet for this repo, but backstage is still periodically receiving events for it.

Reply:
ops sorry @davelil4 I missed answering this thread 🙈
So if the repository isn't found I guess `response` will answer with a `404` right? If yes, maybe we could do:

```ts
if(response.ok) {
  return response.json();
}
throw await ResponseError.fromResponse(response);
```

in this case we are able to map 1:1 the status contained in `response`. Then of course if you want to return a more informative message in case of element not found you could do:

```ts
if(response.ok) {
  return response.json();
}
if(response.status === 404) {
  throw new NotFoundError(`Repository '${options.repo}' in project '${options.projectKey}' doesn't not exist.`):
}
throw await ResponseError.fromResponse(response);
```

so that we remove the "Malformed json response" from the message as it wasn't relevant for the user.

Reply:
I'll include this @vinzscam.

Reply:
Included in newest commut.